src/utils/ucc_parser.c

```diff
@@ -101,3 +93,41 @@ int ucc_config_names_search(ucc_config_names_array_t *config_names,
 
     return -1;
 }
+
+ucc_status_t ucc_config_allow_list_process(const ucc_config_allow_list_t * list,
+                                           const ucc_config_names_array_t *all,
+                                           ucc_config_allow_list_t *       out)
+{
+    ucc_status_t status;
+    int          i;
+
+    out->mode        = list->mode;
+    out->array.names = NULL;
+
+    if (out->mode == UCC_CONFIG_ALLOW_LIST_ALLOW) {
+        status = ucc_config_names_array_dup(&out->array, &list->array);
+        if (UCC_OK != status) {
+            ucc_error("failed to dup config_names_array");
+            return status;
+        }
+    } else {
+        ucc_assert(out->mode == UCC_CONFIG_ALLOW_LIST_NEGATE ||
+                   out->mode == UCC_CONFIG_ALLOW_LIST_ALLOW_ALL);
+
+        out->array.count = 0;
+        out->array.names = ucc_malloc(sizeof(char *) * all->count, "names");
+        if (!out->array.names) {
+            ucc_error("failed to allocate %zd bytes for names array",
+                      sizeof(char *) * all->count);
+            return UCC_ERR_NO_MEMORY;
+        }
+
+        for (i = 0; i < all->count; i++) {
+            if (out->mode == UCC_CONFIG_ALLOW_LIST_ALLOW_ALL ||
+                (ucc_config_names_search(&list->array, all->names[i]) < 0)) {
```

Comment:
is out allow_list mode correct here? even if input list mode is NEGATE out list mode should be ALLOW anyway

Reply:
the idea was that i just copy "mode" to out list and the only value i care about is mode = "ALLOW". This means the list is explicitly requested by user. Other 2 modes: negate or allow_all essentially have no special meaning. We have set of tls from user, we have set of ALL tls and we build another list from these 2.

Reply:
after discussion with Sergey changed to the new helper data struct with explicit "requested" field